src/retyped/codegen.re

```diff
@@ -346,22 +307,35 @@ let rec declaration_to_code module_id types =>
     }
   | InterfaceDecl id type_of =>
     Render.typeDeclaration
-      name::(String.uncapitalize_ascii id)
-      type_of::(bstype_to_code type_of)
-      ();
+      name::(String.uncapitalize_ascii id) type_of::(bstype_to_code type_of) ();
 
 let program_to_code program =>
   switch program {
   | ModuleDecl id statements =>
     let typeof_table = Typetable.create statements;
+    /* is the module nested ? */
```

Comment:
Here we go:

* look for `/`-delimited name
* create `module Whatever = {`/`}` wrapper and apply to code if present